backend/dataPreprocessor.js

```diff
@@ -0,0 +1,93 @@
+const { PrismaClient } = require("@prisma/client");
+const prisma = new PrismaClient();
+const fs = require("fs");
+const natural = require("natural");
+const { TfIdf } = natural;
+const tfidf = new TfIdf();
+const tf = require("@tensorflow/tfjs-node");
+
+// Fetch data from Prisma
+const fetchData = async () => {
+  return await prisma.interaction.findMany({
+    include: {
+      User: true,
+      Business: true,
+    },
+  });
+};
+
+// Handle missing values
+const handleMissingValues = (data) => {
+  return data.map((item) => {
+    item.rated = item.rated !== null ? item.rated : 0;
```

Comment:
Do we want missing values to be handled this way? I think this was a similar situation with your first project where null reviews were rated as 0. 

Generally, I think we should avoid using any non-standard data and filter out data that have missing values.

Reply:
I agree with this, you might get unexpected results if you treat non-rated items as 0.